test/e2e/network/service.go

```diff
@@ -2868,15 +2888,18 @@ var _ = SIGDescribe("Services", func() {
 		framework.ExpectEqual(endpointSubsetOnePorts.Port, int32(8080), "failed to patch Endpoint")
 
 		ginkgo.By("deleting the Endpoint by Collection")
-		err = f.ClientSet.CoreV1().Endpoints(ns).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "testendpoint-static=true"})
+		err = f.ClientSet.CoreV1().Endpoints(testNamespaceName).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{LabelSelector: "test-endpoint-static=true"})
 		framework.ExpectNoError(err, "failed to delete Endpoint by Collection")
 
 		ginkgo.By("waiting for Endpoint deletion")
+		eventFound = false
 		for watchEvent := range endpointWatchChan {
-			if watchEvent.Type == "DELETED" {
+			if watchEvent.Type == watch.Deleted {
+				eventFound = true
 				break
 			}
 		}
+		framework.ExpectEqual(eventFound, true, "failed to find Endpoints %v event", watch.Deleted)
```

Comment:
So we're verifying that a deleted event has occurred.  Is this the same thing as verifying that the Endpoints object has been deleted?  I would suggest trying to retrieve the Endpoints to verify it's no longer there as demonstrated in https://github.com/kubernetes/kubernetes/pull/92621/commits/225e7c75b56cfbdd69d1420a2b42ecee0eefc8e4#diff-cfb4fba63c5d60607f76e0c2e1266d77R225-R231

Reply:
Updated in 95e41663d7ef330874fad713e664b11e1ca320cb